From 1ffe1710cbd632d9f465139ef170f6d0a82129d7 Mon Sep 17 00:00:00 2001
From: Richard Zhu <hongxing.zhu@nxp.com>
Date: Fri, 22 Jun 2018 14:23:57 +0800
Subject: [PATCH 4093/5242] MLK-18660-1 include: define the pd and lpcg of the
 lsio mu

commit  273476349851839fe7bb02e84359dafb097abc04 from
https://source.codeaurora.org/external/imx/linux-imx.git

In order to replace the M4_MU# by the LSIO MU in the
RPMSG usage.
Define the PD and LPCG address of the LSIO MU for iMX8.

Signed-off-by: Richard Zhu <hongxing.zhu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 include/dt-bindings/clock/imx8qm-clock.h  |    7 ++++++-
 include/dt-bindings/clock/imx8qxp-clock.h |    5 ++++-
 include/dt-bindings/soc/imx8_pd.h         |    2 ++
 include/soc/imx8/imx8qm/lpcg.h            |    9 +++++++++
 include/soc/imx8/imx8qxp/lpcg.h           |    9 +++++++++
 5 files changed, 30 insertions(+), 2 deletions(-)

diff --git a/include/dt-bindings/clock/imx8qm-clock.h b/include/dt-bindings/clock/imx8qm-clock.h
index c117867..662cf32 100644
--- a/include/dt-bindings/clock/imx8qm-clock.h
+++ b/include/dt-bindings/clock/imx8qm-clock.h
@@ -844,6 +844,11 @@
 #define IMX8QM_HDMI_RX_HD_CORE_SEL			786
 #define IMX8QM_HDMI_RX_DIG_PLL_CLK			787
 
-#define IMX8QM_CLK_END					788
+#define IMX8QM_LSIO_MU5A_IPG_S_CLK			788
+#define IMX8QM_LSIO_MU5A_IPG_CLK			789
+#define IMX8QM_LSIO_MU6A_IPG_S_CLK			790
+#define IMX8QM_LSIO_MU6A_IPG_CLK			791
+
+#define IMX8QM_CLK_END					792
 
 #endif /* __DT_BINDINGS_CLOCK_IMX8QM_H */
diff --git a/include/dt-bindings/clock/imx8qxp-clock.h b/include/dt-bindings/clock/imx8qxp-clock.h
index 1022fa2..5a101d0 100644
--- a/include/dt-bindings/clock/imx8qxp-clock.h
+++ b/include/dt-bindings/clock/imx8qxp-clock.h
@@ -586,5 +586,8 @@
 #define IMX8QXP_MIPI0_PWM_CLK					526
 #define IMX8QXP_MIPI1_PWM_CLK					527
 
-#define IMX8QXP_CLK_END						528
+#define IMX8QXP_LSIO_MU5A_IPG_S_CLK		528
+#define IMX8QXP_LSIO_MU5A_IPG_CLK		529
+
+#define IMX8QXP_CLK_END						530
 #endif /* __DT_BINDINGS_CLOCK_IMX8QXP_H */
diff --git a/include/dt-bindings/soc/imx8_pd.h b/include/dt-bindings/soc/imx8_pd.h
index 1ab1aeb..0922c9c 100644
--- a/include/dt-bindings/soc/imx8_pd.h
+++ b/include/dt-bindings/soc/imx8_pd.h
@@ -102,6 +102,8 @@
 #define PD_LSIO_PWM_5               lsio_pwm5
 #define PD_LSIO_PWM_6               lsio_pwm6
 #define PD_LSIO_PWM_7               lsio_pwm7
+#define PD_LSIO_MU5A                lsio_mu5a
+#define PD_LSIO_MU6A                lsio_mu6a
 
 #define PD_CONN                     connectivity_power_domain
 #define PD_CONN_SDHC_0              conn_sdhc0
diff --git a/include/soc/imx8/imx8qm/lpcg.h b/include/soc/imx8/imx8qm/lpcg.h
index a16a74a..c41d54d 100644
--- a/include/soc/imx8/imx8qm/lpcg.h
+++ b/include/soc/imx8/imx8qm/lpcg.h
@@ -35,6 +35,15 @@
 #define		GPT_4_LPCG		0x5D580000
 #define		OCRAM_LPCG		0x5D590000
 #define		KPP_LPCG		0x5D5A0000
+#define		MU_5A_LPCG		0x5D600000
+#define		MU_6A_LPCG		0x5D610000
+#define		MU_7A_LPCG		0x5D620000
+#define		MU_8A_LPCG		0x5D630000
+#define		MU_9A_LPCG		0x5D640000
+#define		MU_10A_LPCG		0x5D650000
+#define		MU_11A_LPCG		0x5D660000
+#define		MU_12A_LPCG		0x5D670000
+#define		MU_13A_LPCG		0x5D680000
 
 /* HSIO SS */
 #define		CRR_5_LPCG		0x5F0F0000
diff --git a/include/soc/imx8/imx8qxp/lpcg.h b/include/soc/imx8/imx8qxp/lpcg.h
index e4cd601..79b774d 100644
--- a/include/soc/imx8/imx8qxp/lpcg.h
+++ b/include/soc/imx8/imx8qxp/lpcg.h
@@ -35,6 +35,15 @@
 #define		OCRAM_LPCG		0x5D590000
 #define		KPP_LPCG		0x5D5A0000
 #define		ROMCP_LPCG		0x5D500000
+#define		MU_5A_LPCG		0x5D600000
+#define		MU_6A_LPCG		0x5D610000
+#define		MU_7A_LPCG		0x5D620000
+#define		MU_8A_LPCG		0x5D630000
+#define		MU_9A_LPCG		0x5D640000
+#define		MU_10A_LPCG		0x5D650000
+#define		MU_11A_LPCG		0x5D660000
+#define		MU_12A_LPCG		0x5D670000
+#define		MU_13A_LPCG		0x5D680000
 
 /* HSIO SS */
 #define		CRR_5_LPCG		0x5F0F0000
-- 
1.7.9.5

